cores/esp8266/Updater.cpp

```diff
@@ -57,6 +57,14 @@ bool UpdaterClass::begin(size_t size, int command) {
     return false;
   }
 
+  if(ESP.getFlashChipRealSize() != ESP.getFlashChipSize()) {
```

Comment:
I think we only need to return an error if `ESP.getFlashChipRealSize() < ESP.getFlashChipSize()`.